src/ui/platforms/desktop/src/pages/room/actions/select_action.ts

```diff
@@ -34,17 +35,24 @@ export class SelectAction<T extends GameEventIdentifiers> extends BaseAction {
         this.presenter.disableActionButton('cancel');
       }
 
+      if (requiredAmount instanceof Array) {
+        requiredAmounts = requiredAmount.sort((a, b) => b - a);
+      } else {
+        requiredAmounts = [requiredAmount];
+      }
+
       const selectedPlayers: PlayerId[] = scopedTargets.length === 1 ? scopedTargets.slice() : [];
       for (const player of selectedPlayers) {
         this.presenter.selectPlayer(this.store.room.getPlayerById(player));
       }
 
       this.presenter.setupPlayersSelectionMatcher(
         (player: Player) =>
-          (scopedTargets.includes(player.Id) && selectedPlayers.length < requiredAmount) ||
+          (scopedTargets.includes(player.Id) && selectedPlayers.length < requiredAmounts[0]) ||
```

Comment:
is that correct?

Reply:
requireAmount will be cast to array requireAmounts;
requiredAmount is a descending number array, 
so requiredAmounts[0] is the bigest number player can select.